templates/shared_services/core-application-gateway/porter.yaml

```diff
@@ -0,0 +1,98 @@
+---
+schemaVersion: 1.0.0
+name: tre-shared-service-application-gateway
+version: 0.1.2
+description: "An Azure Application Gateway shared service"
+dockerfile: Dockerfile.tmpl
+registry: azuretre
+
+credentials:
+  - name: azure_tenant_id
+    env: ARM_TENANT_ID
+  - name: azure_subscription_id
+    env: ARM_SUBSCRIPTION_ID
+  - name: azure_client_id
+    env: ARM_CLIENT_ID
+  - name: azure_client_secret
+    env: ARM_CLIENT_SECRET
+
+parameters:
+  - name: tre_id
+    type: string
+    description: "The ID of the parent TRE instance e.g., mytre-dev-3142"
+  - name: id
+    type: string
+    description: "Resource ID"
+  - name: tfstate_resource_group_name
+    type: string
+    description: "Resource group containing the Terraform state storage account"
+  - name: tfstate_storage_account_name
+    type: string
+    description: "The name of the Terraform state storage account"
+  - name: tfstate_container_name
+    type: string
+    default: "tfstate"
+    description: "The name of the Terraform state storage container"
+  - name: arm_use_msi
+    env: ARM_USE_MSI
+    type: boolean
+    default: false
+  - name: backend_collection
+    type: string
+    default: "W10=" # b64 for []
+    description: "Backends list used to route traffic into specific TRE services."
+  - name: arm_environment
+    type: string
+
+mixins:
+  - terraform:
+      clientVersion: 1.4.6
+  - exec
+
+install:
+  # TODO: delete when we use Terraform 1.5+
+  - exec:
+      command: bash
+      arguments:
+        - ./import.sh
```

Comment:
But of an aside, but could we do this with the firewall shared service? I was actually thinking about this this week, will this affect the RP prior to the Firewall being deployed: https://azure.microsoft.com/en-gb/updates/default-outbound-access-for-vms-in-azure-will-be-retired-transition-to-a-new-method-of-internet-access/

Reply:
About the firewall, do you mean install it in core and then import when the shared service is installed?
I'd feel more comfortable with doing the import with terraform 1.5 rather than the script I have now.

Reply:
Yes, from what I understand in that blog post,  and VMs in core won't have internet access until the firewall is deployed.

What is stopping us moving to 1.5?

Reply:
That vm change is 1-2 years away so maybe not pressing to tend to.

Upgrade to 1.5 (or 1.6 as it's released already) would require testing everything. Not a major thing but still. We'll also need to verify the new license.